### PR TITLE
[NFC] Renames a template parameter to avoid clashes with userspace names.

### PR DESCRIPTION
**Pull request by bgra8**

### CODE REVIEW

File: libcxx/include/__format/format_arg_store.h [hunk 1]
```diff
@@ -228,15 +228,15 @@ _LIBCPP_HIDE_FROM_ABI void __store_basic_format_arg(basic_format_arg<_Context>*
   ([&] { *__data++ = __format::__create_format_arg<_Context>(__args); }(), ...);
 }
 
-template <class _Context, size_t N>
+template <class _Context, size_t _Nargs>
```
Comment (mordante) [inline libcxx/include/__format/format_arg_store.h:231]: This is our typical name for these.
```suggestion
template <class _Context, size_t _Np>
```

**Review by mordante — CHANGES_REQUESTED** [review]
Thanks for catching this!

Can you add `N` to https://github.com/llvm/llvm-project/blob/main/libcxx/test/libcxx/system_reserved_names.gen.py#L134 too? That way we will not repeat this mistake.

**Comment by bgra8** [comment]
> Thanks for catching this!
> 
> Can you add `N` to https://github.com/llvm/llvm-project/blob/main/libcxx/test/libcxx/system_reserved_names.gen.py#L134 too? That way we will not repeat this mistake.

Done.

**Review by mordante — APPROVED** [review]
Thanks LGTM!
Can you rebase the patch on main to see whether it passes the CI. The issues look like something I just fixed. When the CI is green feel free to land this patch.

**Comment by philnik777** [comment]
I've just created #76888 to fix more of these issues.

**Comment by bgra8** [comment]
The failing bots have been cancelled for some reason. I see no actual error or test failure.

**Pull request merged**